tools/cli/domain.go

```diff
@@ -45,51 +45,52 @@ func checkRequiredDomainDataKVs(domainData map[string]string) error {
 	return nil
 }
 
-func newDomainCommands() []cli.Command {
-	return []cli.Command{
+func newDomainCommands() []*cli.Command {
+	return []*cli.Command{
 		{
 			Name:    "register",
 			Aliases: []string{"re"},
 			Usage:   "Register workflow domain",
 			Flags:   registerDomainFlags,
-			Action: func(c *cli.Context) {
-				newDomainCLI(c, false).RegisterDomain(c)
+			Action: func(c *cli.Context) error {
+				return newDomainCLI(c, false).RegisterDomain(c)
 			},
 		},
 		{
 			Name:    "update",
 			Aliases: []string{"up", "u"},
 			Usage:   "Update existing workflow domain",
 			Flags:   updateDomainFlags,
-			Action: func(c *cli.Context) {
-				newDomainCLI(c, false).UpdateDomain(c)
+			Action: func(c *cli.Context) error {
+				return newDomainCLI(c, false).UpdateDomain(c)
 			},
 		},
 		{
 			Name:    "deprecate",
 			Aliases: []string{"dep"},
 			Usage:   "Deprecate existing workflow domain",
 			Flags:   deprecateDomainFlags,
-			Action: func(c *cli.Context) {
-				newDomainCLI(c, false).DeprecateDomain(c)
+			Action: func(c *cli.Context) error {
+				return newDomainCLI(c, false).DeprecateDomain(c)
 			},
 		},
 		{
 			Name:    "describe",
 			Aliases: []string{"desc"},
 			Usage:   "Describe existing workflow domain",
 			Flags:   describeDomainFlags,
-			Action: func(c *cli.Context) {
-				newDomainCLI(c, false).DescribeDomain(c)
+			Action: func(c *cli.Context) error {
+				return newDomainCLI(c, false).DescribeDomain(c)
 			},
 		},
 		{
 			Name:    "migration",
 			Aliases: []string{"mi"},
 			Usage:   "Migrate existing domain to new domain. This command only validates the settings. It does not perform actual data migration",
 			Flags:   migrateDomainFlags,
-			Action: func(c *cli.Context) {
+			Action: func(c *cli.Context) error {
 				NewDomainMigrationCommand(c).Validation(c)
+				return nil
```

Comment:
seems like this might be a fairly relevant one to change... though I see `Validation` doesn't return an error.

have you given it a try?  seems fine to leave, just also seems likely to get forgotten.